### PR TITLE
Improve focus distance precision for Nikon Z

### PR DESCRIPTION
**Pull request by kmilos**
This now aligns w/ exiftool as well: https://github.com/exiftool/exiftool/blob/d35e9e26e0a8b443dae307f55d0a4a067d311a16/lib/Image/ExifTool/Nikon.pm#L5363-L5364

### CODE REVIEW

File: src/nikonmn_int.cpp [hunk 1]
```diff
@@ -3963,4 +3963,20 @@ std::ostream& Nikon3MakerNote::printFocalLd4(std::ostream& os, const Value& valu
...
+    os << "(" << value << ")";
+    os.flags(f);
+    return os;
+  }
+  double dist = 0.01 * pow(10.0, (value.toInt64() / 256.0) / 40.0);
+  std::ostringstream oss;
+  oss.copyfmt(os);
+  os << std::fixed << std::setprecision(2) << dist << " m";
+  os.copyfmt(oss);
+  os.flags(f);
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:3978]: I think `copyfmt` saves the flags too, so I think the `flags` code is redundant.

Comment (kmilos) [reply src/nikonmn_int.cpp:3978]: I just did copy/paste of the legacy function, but any new ones indeed omit these, I'll change shortly.

File: src/nikonmn_int.cpp [hunk 2]
```diff
@@ -3963,4 +3963,18 @@ std::ostream& Nikon3MakerNote::printFocalLd4(std::ostream& os, const Value& valu
   return os;
 }
 
+std::ostream& Nikon3MakerNote::printFocusDistanceLd4(std::ostream& os, const Value& value, const ExifData*) {
+  std::ios::fmtflags f(os.flags());
```
Comment (kevinbackhouse) [inline src/nikonmn_int.cpp:3967]:
```suggestion
```

Comment (kmilos) [reply src/nikonmn_int.cpp]: Done, thanks.

**Comment by kmilos** [comment]
Not sure why the test is still reading a byte instead of a short though...

**Review by kevinbackhouse — APPROVED** [review]

**Comment by kmilos** [comment]
@Mergifyio backport 0.28.x

**Pull request merged**